src/rgw/rgw_cache.h

```diff
@@ -26,6 +36,139 @@ enum {
 #define CACHE_FLAG_MODIFY_XATTRS  0x08
 #define CACHE_FLAG_OBJV           0x10
 
+#define mydout(v) lsubdout(T::cct, rgw, v)
+
+/*engage1*/
+struct DataCache;
+class L2CacheThreadPool;
+class HttpL2Request;
+
+struct ChunkDataInfo : public LRUObject {
+	CephContext *cct;
+	uint64_t size;
+	time_t access_time;
+	string address;
+	string oid;
+	bool complete;
+	struct ChunkDataInfo *lru_prev;
+	struct ChunkDataInfo *lru_next;	
+
+	ChunkDataInfo(): size(0) {}
+
+	void set_ctx(CephContext *_cct) {
+		cct = _cct;
+	}
+
+	void dump(Formatter *f) const;
+	static void generate_test_instances(list<ChunkDataInfo*>& o);
+};
+
+struct cacheAioWriteRequest{
+	string oid;
+	void *data;
+	int fd;
+	struct aiocb *cb;
+	DataCache *priv_data;
+	CephContext *cct;
+	
+	cacheAioWriteRequest(CephContext *_cct) : cct(_cct) {}
+	int create_io(bufferlist& bl, unsigned int len, string oid);
+	
+	void release() {
+		::close(fd);
+		cb->aio_buf = NULL;
+		free(data);
+		data = NULL;
+		free(cb);
+		free(this);
+	}
+};
+
+struct DataCache {
+
+private:
+  std::map<string, ChunkDataInfo*> cache_map;
+  std::list<string> outstanding_write_list;
+  int index;
+  RWLock lock;
+  ceph::mutex cache_lock;
+  ceph::mutex req_lock;
+  ceph::mutex eviction_lock;
```

Comment:
similar to the above error also occures at
https://github.com/ceph/ceph/blob/90761e0023abfa54b1b1dcd9feb66fd63c9938ff/src/rgw/rgw_cache.cc#L399
please consider similar instead:
```
  ceph::mutex lock = ceph::make_mutex("DataCache");
  ceph::mutex cache_lock = ceph::make_mutex("DataCache::Mute");
  ceph::mutex req_lock = ceph::make_mutex("DataCache::req");
  ceph::mutex eviction_lock = ceph::make_mutex("DataCache::EvictionMutex");
```